pkg/register/restResolver.go

```diff
@@ -110,6 +105,14 @@ func (c *RestResolverClient) GetDocument(documentID string) (*RegisterDocument,
 		return nil, &ResolverError{err: err, errType: ServerError}
 	}
 
+	if response.StatusCode != http.StatusOK {
```

Comment:
Ah right, my mistake - I think I had moved the condition up higher.
(minor) Worth a minimal comment against the `ReadAll` e.g.: "For certain error cases body of response can contain additional information"